projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.spec.ts

```diff
@@ -166,14 +201,74 @@ describe('LinkValueComponent', () => {
         }
       );
 
-      testHostComponent.inputValueComponent.searchByLabel('thing');
-      testHostFixture.detectChanges();
+      // simulate user searching for label 'thing'
+      testHostComponent.inputValueComponent.valueFormControl.setValue('thing');
 
       expect(valuesSpy.v2.search.doSearchByLabel).toHaveBeenCalledWith('thing', 0, { limitToResourceClass: 'http://0.0.0.0:3333/ontology/0001/anything/v2#Thing'});
       expect(testHostComponent.inputValueComponent.resources.length).toEqual(1);
       expect(testHostComponent.inputValueComponent.resources[0].id).toEqual('http://rdfh.ch/0001/IwMDbs0KQsaxSRUTl2cAIQ');
     });
 
+    it('should not return an invalid update value (string)', () => {
```

Comment:
we have added all the validation functions to only allow input to be an instance of ReadResource, neither in update mode nor in the create mode, the input can be a string. Hence, the update function can never ever be called, we have guaranteed this with validators and there are tests for validators. This test, in my opinion, is completely unnecessary.

Reply:
this test makes sure that `getUpdatedValue` returns false if the form is invalid.

Reply:
This kind of test (to check getUpdateValue returns false if the form is invalid) should be done for all components, not just for the link value component.